pytorch_lightning/trainer/training_loop.py

```diff
@@ -141,6 +141,39 @@ def setup_training(self):
         # --------------------------
         ref_model = self.trainer.get_model()
 
+        # set the ranks and devices
+        self.trainer.accelerator_backend.dist.rank = self.trainer.global_rank
+        self.trainer.accelerator_backend.dist.device = ref_model.device
+
+        # give model convenience properties
+        ref_model.trainer = self.trainer
+
+        # set local properties on the model
+        self.trainer.model_connector.copy_trainer_model_properties(ref_model)
+
+        # init amp. Must be done here instead of __init__ to allow ddp to work
+        if (
+            self.trainer.amp_backend == AMPType.NATIVE and self.trainer.precision == 16
+            and self.trainer._device_type != DeviceType.TPU
+        ):
+            self.trainer.scaler = self.trainer.precision_connector.backend.scaler
+
+        # log hyper-parameters
+        if self.trainer.logger is not None:
+            # save exp to get started (this is where the first experiment logs are written)
+            self.trainer.logger.log_hyperparams(ref_model.hparams_initial)
+            self.trainer.logger.log_graph(ref_model)
+            self.trainer.logger.save()
+
+        # wait for all to join if on distributed
+        self.trainer.accelerator_backend.barrier("setup_training")
+
+        # register auto-resubmit when on SLURM
+        self.trainer.slurm_connector.register_slurm_signal_handlers()
+
+        # --------------------------
+        # Pre-train
+        # --------------------------
```

Comment:
I think this was incorrectly merged here in the last commit. Shouldn't be here right @rohitgr7 ?

Reply:
yes shouldn't be here, should be in `setup_trainer`. will update in few hours.

Reply:
Removed it.